Tasks/JavaToolInstaller/task.json

```diff
@@ -0,0 +1,155 @@
+{
+    "id": "C0E0B74F-0931-47C7-AC27-7C5A19456A36",
+    "name": "JavaToolInstaller",
+    "friendlyName": "Java Tool Installer",
+    "description": "Acquires a specific version of Java from a user supplied Azure blob or the tools cache and adds it to the PATH. Use this step to change the version of Java used in the Java steps.",
+    "helpMarkDown": "[More Information]()",
+    "category": "Tool",
+    "runsOn": [
+        "Agent",
+        "DeploymentGroup"
+    ],
+    "author": "Microsoft Corporation",
+    "version": {
+        "Major": 0,
+        "Minor": 1,
+        "Patch": 0
+    },
+    "preview": true,
+    "satisfies": ["Java"],
+    "demands": [],
+    "groups": [
+        {
+            "name":"JavaInAzureGroup",
+            "displayName": "Download Java from Azure",
+            "isExpanded": true
+        }
+    ],
+    "instanceNameFormat": "Use Java $(versionSpec)",
+    "inputs": [
+        {
+            "name": "versionSpec",
+            "type": "string",
+            "label": "JDK Version",
+            "defaultValue": "1.8.0",
+            "required": true,
+            "helpMarkDown": "A number that specifies the JDK version to make available on the path."
+        },
+        {
+            "name": "jdkSource",
+            "type": "pickList",
+            "label": "JDK source",
+            "required": true,
+            "options": {
+                "AzureStorage": "Azure Storage",
+                "LocalDirectory": "Local Directory"
+            },
+            "helpMarkDown": "Source for the compressed JDK."
+        },
+        {
+            "name": "jdkPath",
+            "type": "filepath",
+            "label": "JDK path",
+            "required": true,
+            "visibleRule": "jdkSource == LocalDirectory",
+            "helpMarkDown": "Path to where the compressed JDK is located. The path could be in your source repository or a local path on the agent."
+        },
+        {
+            "name": "ConnectedServiceNameARM",
```

Comment:
We start these names in general with lower case. To make these more readable in YAML, we should use better names. We also want to use "endpoint" instead of "connectedService". "azureResourceManagerEndpoint"?